dart/lib/src/scope.dart

```diff
@@ -153,6 +161,10 @@ class Scope {
       clone.addBreadcrumb(breadcrumb);
     }
 
+    for (final eventProcessor in _eventProcessors) {
```

Comment:
Since we're in the class, what about a private constructor or factory method that takes all fields as named args?
Then we could just clone all lists etc like `Scope({eventProcessors = [...this.eventprocessors])` and avoid tall these `for`

Reply:
mm yeah but rethinking this right now, maybe it'd make sense to just do a copyWith and follow the convention? tbh I just added the missing field in the existent method, I will check whats the state of the art for cloning objects on Dart

Reply:
https://pub.dev/documentation/copyable/latest/#examples
using the copyable interface seems to be the standard

Reply:
I will add this as a task on our project, so I can merge this now

Reply:
copyWith is probably the "most standard way" ( used a lot in Flutter source code, and in https://pub.dev/packages/freezed, a popular immutable package)